File: VEditModeTransformer.m
//
//  VEditModeTransformer.m
//  tablution
//
//  Created by Ian Mccowan on 6/13/12.
//  Copyright (c) 2012 Nuance, Inc. All rights reserved.
//

#import "VEditModeTransformer.h"

@implementation VEditModeTransformer

+ (Class)transformedValueClass { return [NSString class]; }

+ (BOOL)allowsReverseTransformation { return YES; }

- (id)transformedValue:(id)value {
    return [value boolValue] ? @"Solo Mode" : @"Chord Mode";
}

- (id)reverseTransformedValue:(id)value {
    return [NSNumber numberWithBool:[value isEqualToString:@"Solo Mode"]];
}

@end

File: VEditModeTransformer.swift
//
//  VEditModeTransformer.swift
//  tablution
//
//  Created by Ian McCowan on 10/11/15.
//
//

import Foundation

@objc(VEditModeTransformer) public class VEditModeTransformer: NSValueTransformer {
    public override class func transformedValueClass() -> AnyClass {
        return NSString.self
    }

    public override class func allowsReverseTransformation() -> Bool {
        return true
    }

    public override func transformedValue(value: AnyObject?) -> AnyObject? {
        guard let unwrappedValue = value as? NSNumber else {
            return value
        }
        let isSoloMode = unwrappedValue.boolValue
        if isSoloMode {
            return "Solo Mode"
        }
        else {
            return "Chord Mode"
        }
    }

    public override func reverseTransformedValue(value: AnyObject?) -> AnyObject? {
        if let strValue = value as? String {
            return (strValue == "Solo Mode")
        }
        return nil
    }

}